xdsl/dialects/transform.py

```diff
@@ -147,6 +147,37 @@ class FailurePropagationModeAttr(
 ]
 
 
+@irdl_op_definition
+class ApplyRegisteredPassOp(IRDLOperation):
+    """
+    See external [documentation](https://mlir.llvm.org/docs/Dialects/Transform/#transformapply_registered_pass-transformapplyregisteredpassop).
+    """
+
+    name = "transform.apply_registered_pass"
+
+    options = prop_def(StringAttr, default_value="")
+    pass_name = prop_def(StringAttr)
+    target = operand_def(TransformHandleType)
+    result = result_def(TransformHandleType)
+
+    def __init__(
+        self,
+        pass_name: str,
+        target: SSAValue,
+        options: str | None = None,
+    ):
+        props = {}
+        props["pass_name"] = pass_name
+        if options is not None:
+            props["options"] = options
+
+        super().__init__(
+            properties=props,
```

Comment:
The parent init handles None values
```suggestion
            properties={
                "pass_name": pass_name,
                "options": options,
            },
```

Reply:
Does this work if options is `str | None`?

Reply:
But I would actually make the type of options `str | StringAttr | None` instead of `str | None`, and do `if isinstance(options, str): options = StringAttr(opions)`